pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -2033,6 +2048,12 @@ func (c *Cloud) ensureSecurityGroup(name string, description string) (string, er
 	return groupID, nil
 }
 
+func (s *AWSCloud) ensureSharedSecurityGroup() (string, error) {
+	sgName := SharedSecurityGroupNamePrefix + s.getClusterName()
```

Comment:
In theory this might not be a valid name, but I think in practice the only limit we risk hitting is the length limit, and that is 255 characters so I think we are OK